frontend/awx/resources/hosts/hooks/useHostsActions.tsx

```diff
@@ -1,42 +1,89 @@
 import { PencilAltIcon, TrashIcon } from '@patternfly/react-icons';
-import { useMemo } from 'react';
+import { useCallback, useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import {
   IPageAction,
   PageActionSelection,
   PageActionType,
   usePageNavigate,
 } from '../../../../../framework';
+import { cannotDeleteResource, cannotEditResource } from '../../../../common/utils/RBAChelpers';
+import { awxAPI } from '../../../common/api/awx-utils';
 import { AwxRoute } from '../../../main/AwxRoutes';
 import { AwxHost } from '../../../interfaces/AwxHost';
-import { IAwxView } from '../../../common/useAwxView';
 import { useDeleteHosts } from './useDeleteHosts';
+import { usePatchRequest } from '../../../../common/crud/usePatchRequest';
+import { useParams } from 'react-router-dom';
 
-export function useHostsActions(view: IAwxView<AwxHost>) {
+export function useHostsActions(
+  onDelete: (host: AwxHost[]) => void,
+  onToggle: (() => Promise<void>) | (() => void)
+) {
   const { t } = useTranslation();
   const pageNavigate = usePageNavigate();
-  const deleteHosts = useDeleteHosts(view.unselectItemsAndRefresh);
+
+  const deleteHosts = useDeleteHosts(onDelete);
+
+  const patchRequest = usePatchRequest<{ enabled: boolean }, AwxHost>();
+
+  const params = useParams<{ id: string; inventory_type: string; host_id: string }>();
+
+  const handleToggleHost: (host: AwxHost, enabled: boolean) => Promise<void> = useCallback(
+    async (host, enabled) => {
+      await patchRequest(awxAPI`/hosts/${host.id.toString()}/`, { enabled });
+      await onToggle();
```

Comment:
@prat98 as we talked lets try to use the `updateItem` to update the value return by the PATCH request. Instead of the `view.refresh`?

See: https://github.com/ansible/ansible-ui/pull/1796